dace/transformation/subgraph/composite.py

```diff
@@ -0,0 +1,150 @@
+# Copyright 2019-2021 ETH Zurich and the DaCe authors. All rights reserved.
+""" This module contains classes that implement a composite
+    Subgraph Fusion - Stencil Tiling Transformation
+"""
+
+
+import dace
+from dace.transformation.subgraph import stencil_tiling
+
+import dace.transformation.transformation as transformation
+from dace.transformation.subgraph import SubgraphFusion, MultiExpansion
+from dace.transformation.subgraph.stencil_tiling import StencilTiling
+import dace.transformation.subgraph.helpers as helpers
+
+from dace import dtypes, registry, symbolic, subsets, data
+from dace.properties import make_properties, Property, ShapeProperty
+from dace.sdfg import SDFG, SDFGState
+from dace.sdfg.graph import SubgraphView
+
+import warnings
+
+@registry.autoregister_params(singlestate=True)
+@make_properties
+class CompositeFusion(transformation.SubgraphTransformation):
+    """ MultiExpansion + SubgraphFusion in one Transformation
+        Additional StencilTiling is also possible as a canonicalizing
+        transformation before fusion.
+    """
+
+    debug = Property(desc="Debug mode", dtype=bool, default = False)
+
+    allow_expansion = Property(desc="Allow MultiExpansion first",
+                               dtype = bool,
+                               default = True)
+
+    allow_tiling = Property(desc="Allow StencilTiling (after MultiExpansion)",
+                            dtype = bool,
+                            default = False)
+
+    transient_allocation = Property(
+        desc="Storage Location to push transients to that are "
+             "fully contained within the subgraph.",
+        dtype=dtypes.StorageType,
+        default=dtypes.StorageType.Default)
```

Comment:
use proper enum properties (like my fix in subgraphfusion). otherwise this crashes vscode.